pkg/provider/onepassword/onepassword.go

```diff
@@ -340,13 +341,24 @@ func (provider *ProviderOnePassword) GetSecret(_ context.Context, ref esv1beta1.
 		return nil, err
 	}
 
-	// handle files
+	// handle Document secret type
 	if item.Category == documentCategory {
-		// default to the first file when ref.Property is empty
+		if strings.HasPrefix(ref.Property, "field/") {
+			return provider.getField(item, ref.Property[6:])
+		}
+		if strings.HasPrefix(ref.Property, "file/") {
+			return provider.getFile(item, ref.Property[5:])
+		}
```

Comment:
This could be extracted into a small function and that way the selection logic is encapsulated. Otherwise, it's repeated four times here. :)

Reply:
fixed